src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -207,63 +197,115 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
-
-        // For the success case, register the following callbacks...
-        attemptToSaveClusterInDb foreach { cluster =>
-          // Notify the auth provider and kick off the cluster creation on the Google Dataproc side
-          completeClusterCreation(userEmail, serviceAccountInfo, googleProject, clusterName, augmentedClusterRequest)
-            // TODO Add logging for when the async portion begins and ends for a cluster
-            .onComplete {
-              case Success(_) =>
-                // Finally, notify the cluster monitor that the cluster has been created
-                clusterMonitorSupervisor ! ClusterCreated(cluster, clusterRequest.stopAfterCreation.getOrElse(false))
-              // TODO If the async portion fails we should report the error to the user, probably by setting
-              // the Cluster status to Error in the DB and adding a ClusterError record.
-              case Failure(_) =>
-                // If we didn't succeed, createGoogleCluster removes resources in Google but
-                // we also need to notify our auth provider that the cluster has been deleted.
-                // We won't wait for that deletion, though.
-                authProvider.notifyClusterDeleted(userEmail, userEmail, googleProject, clusterName)
-            }
-        }
+        stageClusterCreation(userEmail, serviceAccountInfo, googleProject, clusterName, clusterRequest)
+    }
+  }
 
-        attemptToSaveClusterInDb
+  private def stageClusterCreation(userEmail: WorkbenchEmail,
+                           serviceAccountInfo: ServiceAccountInfo,
+                           googleProject: GoogleProject,
+                           clusterName: ClusterName,
+                           clusterRequest: ClusterRequest): Future[Cluster] = {
+
+    val augmentedClusterRequest = addClusterLabels(
+      serviceAccountInfo, googleProject, clusterName, userEmail, clusterRequest)
+    val machineConfig = MachineConfigOps.create(clusterRequest.machineConfig, clusterDefaultsConfig)
+    val autopauseThreshold = calculateAutopauseThreshold(
+      clusterRequest.autopause, clusterRequest.autopauseThreshold)
+    val initialCluster = Cluster.create(
+      augmentedClusterRequest, userEmail, clusterName, googleProject,
+      serviceAccountInfo, machineConfig, dataprocConfig.clusterUrlBase, autopauseThreshold)
+
+    // Validate that the Jupyter extension URIs and Jupyter user script URI are valid URIs and reference real GCS objects
+    // and if so, save the cluster creation request parameters in DB
+    val attemptToSaveClusterInDb: Future[Cluster] = validateClusterRequestBucketObjectUri(userEmail, googleProject, augmentedClusterRequest)
+      .flatMap { _ => dbRef.inTransaction(_.clusterQuery.save(initialCluster)) }
+
+    // For the success case, register the following callbacks...
+    attemptToSaveClusterInDb foreach { cluster =>
```

Comment:
Should the `logger.info`s below be `logger.debug`? They were certainly useful during development but I'm not sure if they may constitute too much verbosity.

Reply:
Good question.. I'm inclined to leave these as INFO since it feels like they will be useful.